llvm/lib/Target/X86/X86TargetMachine.cpp

```diff
@@ -117,10 +117,18 @@ static std::unique_ptr<TargetLoweringObjectFile> createTLOF(const Triple &TT) {
   return std::make_unique<X86ELFTargetObjectFile>();
 }
 
-static std::string computeDataLayout(const Triple &TT) {
+static std::string computeDataLayout(const Triple &TT,
+                                     const TargetOptions &Options) {
   // X86 is little endian
   std::string Ret = "e";
 
+  // Decimal floating-point encoding.
+  if (Options.DFPEncoding != BinaryEncoding::None) {
+    assert(Options.DFPEncoding == BinaryEncoding::BID &&
+           "Decimal floating-point on x86 is BID!");
+    Ret += "-d:bid";
+  }
+
```

Comment:
There are multiple places where these strings have to be built for each target? That seems very unfortunate.